lib/generator.js

```diff
@@ -166,9 +162,9 @@ class Generator {
     this.templateContentDir = path.resolve(this.templateDir, TEMPLATE_CONTENT_DIRNAME);
     this.configNunjucks();
     await this.loadTemplateConfig();
+    validateTemplateConfig(this.templateConfig, this.templateParams, asyncapiDocument);
```

Comment:
I changed the order of calling validation and removed it from here https://github.com/asyncapi/generator/pull/349/files#diff-b58e2f1cee2acff55355343774e6d576L646 as I could not come up with an explanation why we were calling validation twice.

also, it is no longer async as I think it should never be